### PR TITLE
[DataGrid] Fix scroll size detector

### LINKED ISSUE #1613: [DataGrid] horizontal scroll behavior is different between light mode and dark mode
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

DataGrid horizontal scroll behavior is different between light mode and dark mode.

https://user-images.githubusercontent.com/35182743/117580602-aa16c700-b133-11eb-8d1b-27ccf870a3a0.mov

## Expected Behavior 🤔

Dark mode behavior should be the same as light mode one.

## Steps to Reproduce 🕹

You can try it here.
https://material-ui.com/components/data-grid/components/

## Context 🔦

I tried to implement dark mode, and found this bug.
The difference may come from here.
https://github.com/mui-org/material-ui-x/blob/master/packages/grid/_modules_/grid/components/containers/GridRootStyles.ts#L331-L363

## Your Environment 🌎
macOS Big Sur version 11.2.3
Google Chrome 90.0.4430.93（Official Build） （x86_64）

### PR DESCRIPTION
**Pull request by dtassone**
fix #1613 

Preview: https://deploy-preview-1703--material-ui-x.netlify.app/components/data-grid/components/

### CODE REVIEW

File: packages/grid/_modules_/grid/hooks/root/useGridContainerProps.ts [hunk 1]
```diff
@@ -71,6 +71,10 @@ export const useGridContainerProps = (
 
       scrollBarSize.y = hasScrollY ? options.scrollbarSize! : 0;
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/hooks/root/useGridContainerProps.ts:72]: This scrollbar behavior impacts both directions. Seeing x and y axis handled differently seems off.

File: packages/grid/_modules_/grid/hooks/root/useGridContainerProps.ts [hunk 2]
```diff
@@ -71,6 +71,10 @@ export const useGridContainerProps = (
...
 
+      // We recalculate the scroll x to consider the size of the y scrollbar.
+      hasScrollX = columnsTotalWidth + scrollBarSize.y > windowSizesRef.current.width;
+      scrollBarSize.x = hasScrollX ? options.scrollbarSize! : 0;
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/hooks/root/useGridContainerProps.ts:76]: As an alternative solution. Did we consider to apply the same CSS to the scrollbar detector as the displayed scrollbar? Could this be more reliable and target the root cause better?

Comment (dtassone) [reply packages/grid/_modules_/grid/hooks/root/useGridContainerProps.ts:76]: There were 2 issues
1. we were off on the calculation of the width of the scrollbar
2. the getScrollBarSize function didn't work properly

**Comment by dtassone** [comment]
> The related bug still happens on mobile. I think we need to apply the CSS only on desktop. I'm using 942px as width in Chrome's Device Mode to test it.
> 
>  scrollbar.mp4

it's because when you change the mode, the state of the scroll bar is not refreshed. You need to reload.
Any idea how I could watch the dark mode switch?

**Comment by m4theushw** [comment]
Even after refreshing the page, the scrollbar is still there. I think the whole idea in changing the color of the scrollbar on dark mode was to fix its look on Windows, because there's no dark scrollbar. However, this causes problems in other platforms. I think we have the following situations:

- On macOS and using the light mode, the scrollbar is invisible by default. It has zero width. Although, when changing to the dark mode, the scrollbar becomes visible and it has to be accounted to calculate the rendering zone width. There's no bug here, it's just the way things work: less space available, put a scrollbar.
- On Windows, the scrollbar is always visible not matter the mode. This way, there's no bug on this platform because the width doesn't change.
- iOS and Android have, by default, a scrollbar that works good on both modes and it's only visible when scrolling, like macOS. Since we change the color on the dark mode, we also force it to become always visible. When there's this change between 
_visible while scrolling_ and _always visible_  the bug occurs.

That being said, I think we have two solutions:

- Apply the CSS to change scrollbar color only on dark mode and **on platforms that don't have a dark scrollbar**. This is a bit hacky, basically we have to check if the user is on Windows.
- Force the scrollbar to be always visible on light mode. AG-Grid does that.

> Any idea how I could watch the dark mode switch?

`React.useEffect`?

**Comment by oliviertassinari** [comment]
> Any idea how I could watch the dark mode switch?

Don't do it, prefer reacting to rerender or listening to change of scrollbar width.

**Comment by dtassone** [comment]
Not sure how you want to fix it now. But it should be fixed one way or another. It doesn't need to be the definitive solution, and we don't need to spend weeks on it...

**Comment by m4theushw** [comment]
@dtassone I would remove the customization for the scrollbar on dark mode. Just noticed that browsers now have a dark scrollbar, but because the docs misses a [`color-scheme`](https://web.dev/color-scheme/) meta tag they don't work. cc @oliviertassinari I presume that if we use the native scrollbar and encourage users to add the tag I mentioned, the issue will be fixed.

Here's an example adding this tag on Edge:

image

**Comment by oliviertassinari** [comment]
@m4theushw Great catch. It looks like we should indeed update `CssBaseline` to take care of this.

**Review by oliviertassinari — APPROVED** [review]

**Comment by oliviertassinari** [comment]
The commit doesn't match the PR title:

Capture d’écran 2021-06-02 à 19 26 54

(*It's more than a refactor, we fix a bug*)

**Pull request merged**